pkg/forwarder/transaction.go

```diff
@@ -142,6 +157,26 @@ func (t *HTTPTransaction) GetTarget() string {
 
 // Process sends the Payload of the transaction to the right Endpoint and Domain.
 func (t *HTTPTransaction) Process(ctx context.Context, client *http.Client) error {
+	t.attemptHandler(t)
```

Comment:
I'm wondering what would be faster here: checking for `nil` instead of calling an empty function or calling it. Not sure how the compiler would optimized this.

If it's the faster or the same I think we should check for nil, that would avoid having to define `defaultAttemptHandler` and `defaultCompletionHandler`.

Reply:
```
BenchmarkEmptyMethod-2   	842194304	         1.32 ns/op	       0 B/op	       0 allocs/op
```

Looks like there is ~1ns of overhead to call an empty method.